common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -24,12 +24,14 @@
 import java.time.temporal.ChronoUnit;
 import java.util.Objects;
 
+import scala.math.Ordered;
+
 import static org.apache.spark.sql.catalyst.util.DateTimeConstants.*;
 
 /**
  * The internal representation of interval type.
  */
-public final class CalendarInterval implements Serializable {
+public final class CalendarInterval implements Serializable, Ordered<CalendarInterval> {
```

Comment:
is it possible to use java orderable? it's a java class

Reply:
I should be able to use java comparable. I will try this.